Signal/src/contact/OWSContactsManager.m

```diff
@@ -191,11 +194,44 @@ - (void)updateSignalAccounts
             [[NSNotificationCenter defaultCenter]
                 postNotificationName:OWSContactsManagerSignalAccountsDidChangeNotification
                               object:nil];
+
+            [self updateCachedDisplayNames];
         });
     });
 }
 
+- (void)updateCachedDisplayNames
+{
+    OWSAssert([NSThread isMainThread]);
+
+    NSMutableDictionary<NSString *, NSString *> *accountNameMap = [NSMutableDictionary new];
+    for (SignalAccount *signalAccount in self.signalAccounts) {
+        NSString *displayName = [self displayNameForSignalAccount:signalAccount];
+        if (![displayName isEqualToString:signalAccount.recipientId]) {
+            accountNameMap[signalAccount.recipientId] = displayName;
+        }
+    }
+
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
We're on the main thread, what's the motivation for dispatching async to the main queue?